### ISSUE TITLE
The image preview node returns to the last interim image when you swap groups

### ISSUE THREAD

**Issue by chrisgoringe** [chrisgoringe/cg-controller#274]

**Comment by chrisgoringe** [comment]
I can't reproduce this - can anyone help?

**Comment by JorgeR81** [comment]
I was going to tell you to do this:

- Open the default workflow 
- Replace `Save Image` with `Image Preview` and add it to a Controller 
- Generate an image
- In the Controler, right click the `Image Preview` to "save as" ( you will get a PNG file ) - **as expected**
- Open another workflow, without closing this one
- Return to the initial workflow In the Controler, 
- Right click the `Image Preview` to "save as" ( you will get a JFIF file, with the Sampler live preview) - **BUG**

---

But I can reproduce it with these steps !

This bug happened in a more complex workflow ( with 5 controllers ). I will try with that and report back.

**Comment by JorgeR81** [comment]
I tried the workflow where I've found the bug, and It worked as expected - **NO BUG**

---

But I think I know what's going on !

It may be related to a different bug. That JFIF file is a "_Fake Image_"

I tried a 3 stage workflow, with stages 2 and 3 bypassed. It worked well. - **NO BUG**

Then I enabled stage 2 ( "Refiner" ), and its `Preview Image` node got filled with the live preview ( JIFIF file ). - _"Fake Image"_
This happened even before I generated any image with that stage. - **different  BUG** 

But after generating an image with stage 2 ( "Refiner" ), it's `Image Preview` got a real PNG image.  
And I can still get it, after switching workflows. - **NO BUG**

---

![b0](https://github.com/user-attachments/assets/19ae7c3d-315b-4690-9385-9211710d5393)
